common/src/main/java/com/sequenceiq/cloudbreak/logger/RestLoggerFilter.java

```diff
@@ -0,0 +1,90 @@
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.TimeZone;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.filter.OncePerRequestFilter;
+import org.springframework.web.util.ContentCachingRequestWrapper;
+import org.springframework.web.util.ContentCachingResponseWrapper;
+
+import com.sequenceiq.cloudbreak.common.anonymizer.AnonymizerUtil;
+
+public class RestLoggerFilter extends OncePerRequestFilter {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private final DateFormat formatter;
+
+    private final boolean restLoggerEnabled;
+
+    public RestLoggerFilter(boolean restLoggerEnabled) {
+        this.restLoggerEnabled = restLoggerEnabled;
+        this.formatter = new SimpleDateFormat("HH:mm:ss.SSS");
+        this.formatter.setTimeZone(TimeZone.getTimeZone("UTC"));
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
+        Date start = new Date(System.currentTimeMillis());
+        ContentCachingRequestWrapper wrappedRequest = getWrappedRequest(request);
+        ContentCachingResponseWrapper wrappedResponse = getWrappedResponse(response);
+
+        filterChain.doFilter(wrappedRequest, wrappedResponse);
+
+        if (restLoggerEnabled) {
+            Date end = new Date(System.currentTimeMillis());
+            String log = new StringBuilder()
+                    .append(appendLine(RestLoggerField.START_TIME, formatter.format(start)))
+                    .append(appendLine(RestLoggerField.END_TIME, formatter.format(end)))
+                    .append(appendLine(RestLoggerField.DURATION, Math.abs(end.getTime() - start.getTime()) + " ms"))
+                    .append(appendLine(RestLoggerField.HTTP_METHOD, request.getMethod()))
+                    .append(appendLine(RestLoggerField.PATH, request.getRequestURI()))
+                    .append(appendLine(RestLoggerField.QUERY_STRING, request.getQueryString()))
+                    .append(appendLine(RestLoggerField.CLIENT_IP, request.getRemoteAddr()))
+                    .append(appendLine(RestLoggerField.REQUEST,
+                            logContent(wrappedRequest.getContentAsByteArray(), request.getCharacterEncoding())))
+                    .append(appendLine(RestLoggerField.RESPONSE_STATUS, String.valueOf(response.getStatus())))
+                    .append(appendLine(RestLoggerField.RESPONSE,
+                            logContent(wrappedResponse.getContentAsByteArray(), request.getCharacterEncoding())))
+                    .toString();
+            logger.debug(log);
```

Comment:
`2020-11-09 15:04:43,771 [http-nio-9091-exec-1] anonymize:34 INFO  c.s.c.c.a.AnonymizerUtil - [type:springLog] [crn:] [name:] [flow:] [requestid:75737203-c895-4fc1-b357-2975722d3b3d] [tenant:] [userCrn:] [environment:] [traceId:] [spanId:] Anonymize the content with length: 2
2020-11-09 15:04:43,772 [http-nio-9091-exec-1] anonymize:34 INFO  c.s.c.c.a.AnonymizerUtil - [type:springLog] [crn:] [name:] [flow:] [requestid:75737203-c895-4fc1-b357-2975722d3b3d] [tenant:] [userCrn:] [environment:] [traceId:] [spanId:] Anonymize the content with length: 15
2020-11-09 15:04:43,773 [http-nio-9091-exec-1] doFilterInternal:62 DEBUG c.s.c.l.RestLoggerFilter - [type:springLog] [crn:] [name:] [flow:] [requestid:75737203-c895-4fc1-b357-2975722d3b3d] [tenant:] [userCrn:] [environment:] [traceId:] [spanId:] startTime: 14:04:26.385 
endTime: 14:04:26.536 
duration: 151 ms 
httpMethod: GET 
path: /cb/health 
queryString: null 
clientIp: 127.0.0.1 
request: {} 
responseStatus: 200 
response: {"status":"UP"} 
`

Unfortunately, thenant Id traceId and such things were not filled out.

Reply:
ok let me reorder the filters mdc must come first